#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Adding empty m68k framestate.c not to build for dwarf2.
# DP: Related bugs: 
# DP: Dpatch author: Philip Blundell <pb@nexus.co.uk>
# DP: Patch author: Philip Blundell <pb@nexus.co.uk>
# DP: Upstream status: Debian-Specific
# DP: Status Details: ARM uses sjlj exceptions, not dwarf2 unwind.
# DP:		This patch adds empty framestate.c for arm to avoid
# DP:		dwarf2 build failure.
# DP: Date: 2003-07-19 (Update 2005-03-14 gotom)

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

2005-03-14  GOTO Masanori  <gotom@debian.org>

	* sysdeps/arm/unwind-pe.c: Update to fix compilation failure.

2003-07-19  GOTO Masanori  <gotom@debian.org>

	* sysdeps/arm/framestate.c: Add to fix compilation failure
	with sjlj exception, suggested by Philip Blundell.


--- /dev/null   2003-01-13 18:16:28.000000000 +0900
+++ sysdeps/arm/framestate.c    2003-07-16 23:21:34.000000000 +0900
@@ -0,0 +1,2 @@
+
+
--- /dev/null	2004-12-05 06:07:48.000000000 +0000
+++ sysdeps/arm/unwind-pe.c	2005-03-14 05:14:16.000000000 +0000
@@ -0,0 +1,6 @@
+#include <stdlib.h>
+#include <unwind.h>
+
+#define _LIBC_DEFINITIONS
+#define NO_BASE_OF_ENCODED_VALUE
+#include "unwind-pe.h"
